### PR TITLE
DPTS-19 Implement iam roles for pods

### PR DESCRIPTION
**Pull request by jope1**
Terraformed service IAM role and attached a "dummy" policy. 
Configured K8s deployment files to use the IAM service role.

### CODE REVIEW

File: build/automation/var/platform-texas/platform-texas-v1.mk [hunk 1]
```diff
@@ -24,6 +24,7 @@ TF_VAR_aws_profile = nhsd-ddc-exeter-texas-live-lk8s-$(AWS_ACCOUNT_NAME)
 TF_VAR_platform_zone = $(TEXAS_HOSTED_ZONE)
 
 TF_VAR_route53_terraform_state_key = route53/terraform.tfstate
+TF_VAR_eks_mgmt_terraform_state_key = eks/terraform.tfstate
```
Comment (dehe1) [inline build/automation/var/platform-texas/platform-texas-v1.mk:27]: Being pedantic but probably should be named "eks" not "eks_mgmt"

File: infrastructure/stacks/service/iam.tf [hunk 2]
```diff
@@ -0,0 +1,54 @@
+# Data lookup from eks management terraform state
+data "terraform_remote_state" "eks-mgmt" {
```
Comment (dehe1) [inline infrastructure/stacks/service/iam.tf:2]: Same here

File: infrastructure/stacks/service/variables.tf [hunk 3]
```diff
@@ -38,6 +41,10 @@ variable "route53_terraform_state_key" {
   description = "The Route 53 key in the terraform state bucket"
 }
 
+variable "eks_mgmt_terraform_state_key" {
```
Comment (dehe1) [inline infrastructure/stacks/service/variables.tf:44]: and here

**Review by dehe1 — CHANGES_REQUESTED** [review]
Just a quick refactor, you're using the eks stack in the k8s account, not the eks mgmt stack in the mgmt account, so a little more readable to edit it back to "eks" not "eks_mgmt"

**Review by dehe1 — APPROVED** [review]
All good, thanks

**Review by timrickwood — APPROVED** [review]

**Review by timrickwood — DISMISSED** [review]

**Review by timrickwood — APPROVED** [review]

**Review by tamaragoldschmidt — APPROVED** [review]

**Pull request merged**